File: 2418. Sort the People/sortPeople.py
from typing import List

class Solution:
  def sortPeople(self, names: List[str], heights: List[int]) -> List[str]:
    heightsToNames = {}
    
    for i in range(len(heights)):
      heightsToNames[heights[i]] = names[i]
    
    res = [] 
    for h in sorted(heights, reverse = True):
      res.append(heightsToNames[h])
    
    return res
    
names = ["Mary","John","Emma"]
heights = [180,165,170]
solution = Solution()
print('sortPeople: ', solution.sortPeople(names, heights))

File: 2418. Sort the People/sortPeople.js
/**
 * @param {string[]} names
 * @param {number[]} heights
 * @return {string[]}
 */
const sortPeople = (names, heights) => {
  const heightsToNames = {};
  for (let i = 0; i < heights.length; i++) {
    heightsToNames[heights[i]] = names[i];
  }

  const res = [];
  heights.sort((a, b) => b - a);
  for (let i = 0; i < heights.length; i++) {
    res.push(heightsToNames[heights[i]]);
  }

  return res;
};


const names = ["Mary", "John", "Emma"];
const heights = [180, 165, 170];
console.log('sortPeople: ', sortPeople(names, heights))
